lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -594,6 +597,100 @@ Return a new mode with its [`ABI`](@ref) set to the chosen type.
 """
 function set_abi end
 
+"""
+    split_mode(
+        ::ReverseMode, [::Val{ReturnShadow}, ::Val{Width}, ::Val{ModifiedBetween}, ::Val{ShadowInit}]
+    )
+
+Turn a [`ReverseMode`](@ref) object into a [`ReverseModeSplit`](@ref) object while preserving as many of the settings as possible.
+The rest of the settings can be configured with optional positional arguments of `Val` type.
+
+This function acts as the identity on a [`ReverseModeSplit`](@ref).
+
+See also [`combined_mode`](@ref).
+"""
+function split_mode(
+    ::ReverseMode{
+        ReturnPrimal,
+        RuntimeActivity,
+        ABI,
+        Holomorphic,
+        ErrIfFuncWritten
+    },
+    ::Val{ReturnShadow}=Val(true),
+    ::Val{Width}=Val(0),
+    ::Val{ModifiedBetween}=Val(true),
+    ::Val{ShadowInit}=Val(false),
+) where {
+    ReturnPrimal,
+    ReturnShadow,
+    RuntimeActivity,
+    Width,
+    ModifiedBetween,
+    ABI,
+    Holomorphic,
+    ErrIfFuncWritten,
+    ShadowInit
+}
+    mode_split = ReverseModeSplit{
+        ReturnPrimal,
+        ReturnShadow,
+        RuntimeActivity,
+        Width,
+        ModifiedBetween,
+        ABI,
+        Holomorphic,
+        ErrIfFuncWritten,
+        ShadowInit
+    }()
+    return mode_split
+end
+
+split_mode(mode::ReverseModeSplit, args...) = mode
+
+"""
+    combined_mode(::ReverseMode)
+
+Turn a [`ReverseModeSplit`](@ref) object into a [`ReverseMode`](@ref) object while preserving as many of the settings as possible.
+
+This function acts as the identity on a [`ReverseMode`](@ref).
+
+See also [`split_mode`](@ref).
+"""
+function combined_mode(
```

Comment:
@vchuravy should we have this be lowercase or upper case. I'm actually now thinking maybe uppercase. I like the WithPrimal(Forward) == ForwardWithPrimal property.

Reply:
so then similarly here Split(ReverseWithPrimal) == ReverseSplitWithPrimal

Reply:
Whenever possible, it would be nice to keep [consistency with Base Julia](https://docs.julialang.org/en/v1/manual/style-guide/#Use-naming-conventions-consistent-with-Julia-base/): uppercase for types, lowercase for functions.
Of course this is already broken by some of the existing mode utilities, and it's too late to go back, but maybe adopting a minimal style guide would help? That way, contributors know right away how to comply, instead of following changes of hearts from maintainers.